src/overlay/Peer.cpp

```diff
@@ -1216,20 +1216,25 @@ Peer::recvAuth(StellarMessage const& msg)
         return;
     }
 
-    // ask for SCP state if not synced
-    // this requests data for slots lcl-window ... latest consensus (if
-    // possible) we need to ask for older slots in case other peers need
-    // messages we didn't need ourself
+    // this requests SCP data for all slots after `low`
+    // we ask for a few older slots before what our own lcl, in case other peers
+    // need messages we didn't need ourself
     auto low = mApp.getLedgerManager().getLastClosedLedgerNum() + 1;
-    auto maxSlots = mApp.getConfig().MAX_SLOTS_TO_REMEMBER;
+    auto maxSlots = std::min<uint32>(mApp.getConfig().MAX_SLOTS_TO_REMEMBER,
+                                     Config::SCP_EXTRA_LOOKBACK_LEDGERS);
     if (low > maxSlots)
```

Comment:
we have this exact same logic in `HerderImpl::getMoreSCPState`, which is called when a node goes out of sync. Should the new calculation be used there as well? (and if so, can we refactor it to avoid duplication?)

Reply:
ah good catch, I didn't see we had this duplicated